File: spring-web/src/main/java/com/example/springwebex/util/HttpHeaderUtil.java
package com.example.springwebex.util;

import java.io.UnsupportedEncodingException;
import java.nio.charset.StandardCharsets;
import java.util.Base64;
import java.util.Enumeration;

public class HttpHeaderUtil {

    private static final String BEARER_TYPE = "Bearer";

    /**
     * get Autorizaion header
     */
    public static String getToken(Enumeration<String> authorization) {
        while (authorization.hasMoreElements()) {
            String value = authorization.nextElement();
            if (StringUtil.toLowerCase(value).startsWith(StringUtil.toLowerCase(BEARER_TYPE))) {
                String authHeaderValue = value.substring(BEARER_TYPE.length()).trim();
                int commaIndex = authHeaderValue.indexOf(',');
                if (commaIndex > 0) {
                    authHeaderValue = authHeaderValue.substring(0, commaIndex);
                }
                return authHeaderValue;
            }
        }
        return null;
    }

    /**
     * get Autorizaion header after base64 decoding
     */
    public static String getBase64DecodedToken(Enumeration<String> authorization) throws UnsupportedEncodingException {
        while (authorization.hasMoreElements()) {
            String value = authorization.nextElement();
            if (StringUtil.toLowerCase(value).startsWith(StringUtil.toLowerCase(BEARER_TYPE))) {
                String authHeaderValue = value.substring(BEARER_TYPE.length()).trim();
                int commaIndex = authHeaderValue.indexOf(',');
                if (commaIndex > 0) {
                    authHeaderValue = authHeaderValue.substring(0, commaIndex);
                }
                return new String(Base64.getDecoder().decode(authHeaderValue.getBytes(StandardCharsets.UTF_8)),
                        StandardCharsets.UTF_8);
            }
        }
        return null;
    }

    /**
     * get Autorizaion header after base64 decoding
     */
    public static String getBase64DecodedToken(String authorization) throws UnsupportedEncodingException {
        if (StringUtil.toLowerCase(authorization).startsWith(StringUtil.toLowerCase(BEARER_TYPE))) {
            String authHeaderValue = authorization.substring(BEARER_TYPE.length()).trim();
            int commaIndex = authHeaderValue.indexOf(',');
            if (commaIndex > 0) {
                authHeaderValue = authHeaderValue.substring(0, commaIndex);
            }
            return new String(Base64.getDecoder().decode(authHeaderValue.getBytes(StandardCharsets.UTF_8)),
                    StandardCharsets.UTF_8);
        }
        return null;
    }

}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/util/HttpHeaderUtil.kt
package com.example.springwebex.util

import java.nio.charset.StandardCharsets
import java.util.*

object HttpHeaderUtil {

    private const val BEARER_TYPE = "Bearer"

    fun getToken(authorization: Enumeration<String>): String? {
        while (authorization.hasMoreElements()) {
            val value = authorization.nextElement()
            if (StringUtil.toLowerCase(value)?.startsWith(StringUtil.toLowerCase(BEARER_TYPE) ?: "") == true) {
                var authHeaderValue = value.substring(BEARER_TYPE.length).trim()
                val commaIndex = authHeaderValue.indexOf(',')
                if (commaIndex > 0) {
                    authHeaderValue = authHeaderValue.substring(0, commaIndex)
                }
                return authHeaderValue
            }
        }
        return null
    }

    fun getBase64DecodedToken(authorization: String?): String? {
        if (authorization != null && StringUtil.toLowerCase(authorization)?.startsWith(StringUtil.toLowerCase(BEARER_TYPE) ?: "") == true) {
            var authHeaderValue = authorization.substring(BEARER_TYPE.length).trim()
            val commaIndex = authHeaderValue.indexOf(',')
            if (commaIndex > 0) {
                authHeaderValue = authHeaderValue.substring(0, commaIndex)
            }
            return String(
                Base64.getDecoder().decode(authHeaderValue.toByteArray(StandardCharsets.UTF_8)),
                StandardCharsets.UTF_8
            )
        }
        return null
    }
}
